src/pub.rs

```diff
@@ -165,6 +165,7 @@ impl SocketSend for PubSocket {
         let mut dead_peers = Vec::new();
         for mut subscriber in self.backend.subscribers.iter_mut() {
             for sub_filter in &subscriber.subscriptions {
+                if sub_filter.len() == message.get(0).unwrap().len(){
```

Comment:
@youngday thanks for the pull request. I agree that this is an issue however IMHO your fix is not quite compatible to ZMQ spec:
"SHALL perform a binary comparison of the subscription against the start of the first frame of the message."
So the proper fix would be to 
sub_filter.len() <= ...